decode_symphonia/src/lib.rs

```diff
@@ -104,13 +102,30 @@ impl Decoder for SymphoniaDecoder {
 
         // Create a decoder for the stream.
         let mut decoder = symphonia::default::get_codecs().make(&params, &decoder_opts)?;
+        debug_assert_eq!(params.n_frames, decoder.codec_params().n_frames);
+        debug_assert_eq!(params.sample_rate, decoder.codec_params().sample_rate);
+        debug_assert_eq!(params.channels, decoder.codec_params().channels);
+
+        // The stream/decoder might not always provide the actual numbers
+        // of channels (MP4/AAC/ALAC). In this case the number of channels
+        // will be obtained from the signal spec of the first decoded packet.
+        let mut channels = params.channels;
 
         // Decode the first packet to get the signal specification.
         let smp_buf = loop {
             match decoder.decode(&reader.next_packet()?) {
                 Ok(decoded) => {
                     // Get the buffer spec.
                     let spec = *decoded.spec();
+                    if let Some(channels) = channels {
+                        debug_assert_eq!(channels, spec.channels);
+                    } else {
+                        println!(
+                            "Assuming {num_channels} channel(s) according to the first decoded packet",
+                            num_channels = spec.channels.count()
+                        );
```

Comment:
unconditionally printing to stdout isn't great
```suggestion
                        log::debug!(
                            "Assuming {num_channels} channel(s) according to the first decoded packet",
                            num_channels = spec.channels.count()
                        );
```

Reply:
I have seen println!() statements everywhere and simply adopted it. Of course, `log::debug` would be more suitable.

Reply:
I do see a few more `println!` statements in the library. Changing those to also use `log::debug` would be nice.

Reply:
...but the `log` dependency is missing and first needs to be discussed. Should be introduced separately if desired and not by this PR.

Reply:
[c644953](https://github.com/MeadowlarkDAW/creek/pull/22/commits/c64495311d5d6efab29bbd46da541fad16e76715)

Reply:
#23